Add doc dir for Keptn 0.16 (#1228)
keptn/keptn.github.io: content/docs/0.16.x/troubleshooting/index.md

---
title: Troubleshooting
description: Find tips and tricks to deal with troubles that may occur when using Keptn. 
weight: 100
icon: help
---

In this section, instructions have been summarized that help to troubleshoot known issues that may occur when using Keptn.

## Generating a Support Archive

> The Keptn CLI allows to generate a support archive, which can be used as data source for debugging a Keptn installation.

Reviewer: ```suggestion
Use the Keptn CLI to generate a support archive, which can be used as a data source for debugging a Keptn installation.
```

<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx/1.17.9</center>
</body>
</html>
```

You can resolve this problem by restarting the Nginx ingress with the following command:

```
$ kubectl -n keptn delete pod -l run=api-gateway-nginx

pod "api-gateway-nginx-cc948646d-zwrb4" deleted
```

After some seconds, the Nginx ingress pod should be up and running again. You can verify this by executing:

```
$ kubectl get pods -n keptn -l run=api-gateway-nginx

NAME                                READY   STATUS    RESTARTS   AGE
api-gateway-nginx-cc948646d-h6bdb   1/1     Running   0          13m
```

At this point, the CLI commands should work again:

```
$ keptn add-resource --project=sockshop --service=carts --stage=production --resource=remediation.yaml

Adding resource remediation.yaml to service carts in stage production in project sockshop
Resource has been uploaded.
```

## Keptn on Minikube causes a MongoDB issue after a reboot

> When rebooting the machine Minikube is installed on, the MongoDB pod in the `keptn` namespace runs in a `CrashLoopBackoff`. 

Reviewer: ```suggestion
When rebooting the machine on which Minikube is installed, the MongoDB pod in the `keptn` namespace runs in a `CrashLoopBackoff`. 
```

<details><summary>Expand instructions</summary>
<p>

> **Note:** Minikube is a K8s distribution for development environments. Please go with K3s/K3d for a more stable setup.

Reviewer: ```suggestion
**Note:** Minikube is a Kubernetes distribution for development environments. Please go with K3s/K3d for a more stable setup.
```

**Analysis**: To find out whether you are affected or not, please run an `alpine:3.11` container that tries to access the Kubernetes API or any external hostname, e.g.:

```
kubectl run -i --restart=Never --rm test-${RANDOM} --image=alpine:3.11 -- sh -c "wget --no-check-certificate https://kubernetes.default.svc.cluster.local/api/v1"
```

```
kubectl run -i --restart=Never --rm test-${RANDOM} --image=alpine:3.11 -- sh -c "wget https://keptn.sh"
```

If in any of the above instances you get a "bad address", then you are most likely affected, e.g.:
```
wget: bad address 'kubernetes.default.svc.cluster.local'
```

If it prints a download bar, the content of the requested URL or an HTTP 400 error (or similar), the connection works, e.g.:
```
Connecting to kubernetes.default.svc.cluster.local (10.0.80.1:443)
saving to 'v1'
v1                   100% |********************************| 10337  0:00:00 ETA
```

The problem behind this is usually a misconfiguration for the nameserver or the local `/etc/resolv.conf` configuration (e.g., searchdomains). 

More details can be found at [GitHub Kubernetes Issue #64924](https://github.com/kubernetes/kubernetes/issues/64924).

**Solutions**: 

> * Verify your clusters nameserver configuration is working as expected, especially the searchdomains. Easiest way to verify is to look at the output of

Reviewer: ```suggestion
* Verify that your cluster's nameserver configuration is working as expected, especially the searchdomains. Easiest way to verify is to look at the output of
```
